### PR TITLE
✨ added new flow for extracting prefect logs

### PR DESCRIPTION
**Pull request by gwieloch**
## Summary

Added new flow that extracts logs from Prefect API within selected date ranges.

## Importance

Prefect stores information on flows statuses only for last 14 days. With PrefectLogs flow we will keep this info stored in ADLS folder.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/flows/prefect_logs.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import prefect
+import pandas as pd
+import datetime
+
+from prefect import Flow, task, Task
+from prefect.utilities import logging
+
+from viadot.tasks.azure_data_lake import AzureDataLakeUpload
+
+from viadot.task_utils import add_ingestion_metadata_task, df_to_parquet
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:11]: If you want to split imports for the groups maybe the better idea is to split by `prefect imports`, `viadot imports`, `other`.
Here there are 4 blank lines that we can reduce.

File: viadot/flows/prefect_logs.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        super().__init__(
+            name="prefect_extract_flows_logs",
+            *args,
+            **kwargs,
+        )
+
+        self.gen_flow()
+
+    def get_formatted_date_from_timestamp(self, col_value):
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:65]: add `type annotation` , `return type` and `docstring`.

File: viadot/flows/prefect_logs.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            scheduled_start_time (str, optional): A parameter passed to the Prefect API query. Defaults to None.
+            filter_type (Literal, optional): A comparison operator passed to the Prefect API query (_gte >=, _lte <=)  Defaults to _gte.
+            local_file_path (str, optional): Local destination path. Defaults to None.
+            adls_path (str, optional): Azure Data Lake destination folder/catalog path. Defaults to None.
+            adls_sp_credentials_secret (str, optional): The name of the Azure Key Vault secret containing a dictionary with
+                ACCOUNT_NAME and Service Principal credentials (TENANT_ID, CLIENT_ID, CLIENT_SECRET) for the Azure Data Lake.
+                Defaults to None.
+            vault_name (str, optional): The name of the vault from which to obtain the secrets. Defaults to None.
+            overwrite_adls (bool, optional): Whether to overwrite the file in ADLS. Defaults to True.
+
```
[the remark this answers is not in the document]
Comment (gwieloch) [reply viadot/flows/prefect_logs.py]: done

File: viadot/flows/prefect_logs.py [hunk 4]
```diff
@@ -33,16 +32,16 @@ def __init__(
 
         Args:
             name (str): The name of the flow.
-            scheduled_start_time (str, optional): A parameter passed to the Prefect API query. Defaults to None.
-            filter_type (Literal, optional): A comparison operator passed to the Prefect API query (_gte >=, _lte <=)  Defaults to _gte.
+            scheduled_start_time (str, optional): A parameter passed to the Prefect API query. Defaults to 'yesterday'.
+            filter_type (Literal, optional): A comparison operator passed to the Prefect API query (_gte >=, _lte <=) that refers to the left or right boundary of date ranges
+                for which flow extracts data. Defaults to _gte.
             local_file_path (str, optional): Local destination path. Defaults to None.
             adls_path (str, optional): Azure Data Lake destination folder/catalog path. Defaults to None.
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:39]: `folder/catalog path` -> `catalog or file path`

File: viadot/flows/prefect_logs.py [hunk 5]
```diff
@@ -54,6 +53,13 @@ def __init__(
...
 
+        if scheduled_start_time == "yesterday":
+            self.scheduled_start_time = datetime.date.today() - datetime.timedelta(
+                days=1
+            )
+        else:
+            self.scheduled_start_time = scheduled_start_time
+
         super().__init__(
             name="prefect_extract_flows_logs",
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:64]: `prefect_extract_flow_logs` - not sure but maybe it would be better

File: viadot/flows/prefect_logs.py [hunk 6]
```diff
@@ -62,34 +68,47 @@ def __init__(
 
         self.gen_flow()
 
-    def get_formatted_date_from_timestamp(self, col_value):
-        if col_value is not None:
-            return col_value.split("T")[0]
+    def get_formatted_date_from_timestamp(self, value) -> str:
+        """
+        Function returns cleaned date from timestamp in string format (ex. "2022-01-01")
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:73]: Add full stop and add blank line after the description

**Review by Rafalz13 — COMMENTED** [review]
Please apply some changes:
* clean `schedule_start_time` -> only time (`hh:mm:ss`)
* add new column base on `created_by_user_id` to show if it was started automatically or not

File: viadot/flows/prefect_logs.py [hunk 7]
```diff
@@ -61,22 +62,34 @@ def __init__(
...
+        Return: (str) Date or time from the Timestamp
         """
-
-        if value is not None:
+        if value is not None and value_type == "date":
             return value.split("T")[0]
+        elif value is not None and value_type == "time":
+            return (value.split("T")[1])[:5]
+
+    def check_if_run_authomatically(self, value) -> bool:
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:85]: add type hint to `value` variable

File: viadot/flows/prefect_logs.py [hunk 8]
```diff
@@ -61,22 +62,34 @@ def __init__(
...
+        """
+        Function checks if flow has been run automatically by scheduler or triggered manually.
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:87]: Update docstring
```
[description]

Args:
   value: ...

Return:
    ....

```

File: viadot/flows/prefect_logs.py [hunk 9]
```diff
@@ -61,22 +62,34 @@ def __init__(
...
             **kwargs,
         )
 
         self.gen_flow()
 
-    def get_formatted_date_from_timestamp(self, value) -> str:
+    def get_formatted_value_from_timestamp(self, value_type, value) -> str:
         """
-        Function returns cleaned date from timestamp in string format (ex. "2022-01-01")
+        Function returns cleaned date/time from timestamp in string format (ex. "2022-01-01")
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:74]: Add a period at the end.

File: viadot/flows/prefect_logs.py [hunk 10]
```diff
@@ -1,3 +1,4 @@
+from sqlite3 import Timestamp
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:1]: Is it a good import? Timestamp from sqlite3?

File: viadot/flows/prefect_logs.py [hunk 11]
```diff
@@ -61,22 +62,34 @@ def __init__(
...
-            name="prefect_extract_flows_logs",
+            name="prefect_extract_flow_logs",
             *args,
             **kwargs,
         )
 
         self.gen_flow()
 
-    def get_formatted_date_from_timestamp(self, value) -> str:
+    def get_formatted_value_from_timestamp(self, value_type, value) -> str:
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:72]: Add type hints to `value_type` and `value`

File: viadot/flows/prefect_logs.py [hunk 12]
```diff
@@ -61,22 +62,34 @@ def __init__(
...
         """
-        Function returns cleaned date from timestamp in string format (ex. "2022-01-01")
+        Function returns cleaned date/time from timestamp in string format (ex. "2022-01-01")
         Args:
+            type (str):  Desired part of the Timestamp value, "date" or "time".
             value (timestamp): Timestamp value from Prefect.
+        Return: (str) Date or time from the Timestamp
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:78]: 1) Add a period at the end.
2) Change to 
```
Return:
    str: Date or .......
```

**Review by Rafalz13 — COMMENTED** [review]
Small changes connected with docstrings and and type hints.

File: viadot/flows/prefect_logs.py (hunk 10 above)
Comment (gwieloch) [reply viadot/flows/prefect_logs.py:1]: changed to 
from pandas import Timestamp

File: viadot/flows/prefect_logs.py [hunk 13]
```diff
@@ -69,22 +69,30 @@ def __init__(
...
 
-    def get_formatted_value_from_timestamp(self, value_type, value) -> str:
+    def get_formatted_value_from_timestamp(
+        self, value_type: str, value: Timestamp
+    ) -> str:
         """
         Function returns cleaned date/time from timestamp in string format (ex. "2022-01-01")
         Args:
-            type (str):  Desired part of the Timestamp value, "date" or "time".
+            value_type (str):  Desired part of the Timestamp value, "date" or "time".
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:78]: Unify data types.
line 73 - you have `Timestamp` type. In line 78 is `str` - which is correct?

File: viadot/flows/prefect_logs.py [hunk 14]
```diff
@@ -69,22 +69,30 @@ def __init__(
...
             value (timestamp): Timestamp value from Prefect.
-        Return: (str) Date or time from the Timestamp
+        Return:
+            str: date (ex. "2022-01-01") or time (ex. "13:10") from the Timestamp in string format
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:81]: Add `.` at the end.

File: viadot/flows/prefect_logs.py [hunk 15]
```diff
@@ -69,22 +69,30 @@ def __init__(
 
         self.gen_flow()
 
-    def get_formatted_value_from_timestamp(self, value_type, value) -> str:
+    def get_formatted_value_from_timestamp(
+        self, value_type: str, value: Timestamp
+    ) -> str:
         """
         Function returns cleaned date/time from timestamp in string format (ex. "2022-01-01")
```
Comment (Rafalz13) [inline viadot/flows/prefect_logs.py:76]: Add `.` at the end.

**Review by Rafalz13 — COMMENTED** [review]
@m-paz
Looks good, only minor changes.

**Review by Rafalz13 — APPROVED** [review]

**Pull request merged**